lib/rubocop/cop/mixin/index_method.rb

```diff
@@ -134,7 +134,7 @@ def self.from_map_to_h(node, match)
         end
 
         def self.from_hash_brackets_map(node, match)
-          new(match, node.children.last, 'Hash['.length, ']'.length)
+          new(match, node.children.last, "#{node.receiver.source}[".length, ']'.length)
```

Comment:
This is a change that was not originally envisioned for the purpose of "Fix false-positives", but after adding tests, I discovered that there is an autocorrection bug where `::Hash[...]` is used.

Should I separate a Pull Request to fix this bug?

Reply:
Yeah, you could do it separately because the automatic correction mistake is unclear in the PR's aim to the context.

Reply:
Thanks, I created the following Pull Request for this autocorrection bug:

- https://github.com/rubocop/rubocop-rails/pull/867